lib/portlayer/storage/compute/stat.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// StatPath will use Guest Tools to stat a given path in the container
+
+package compute
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/govmomi/guest"
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/lib/portlayer/exec"
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/pkg/errors"
+)
+
+// TODO: still need to figure this part out
+type FileStat struct {
+	LinkTarget string
+	Mode       uint32
+	Name       string
+	Size       int64
+}
+
+// interface for offline container statpath
+type ContainerStatPath interface {
+	// deviceId, filepath
+	StatPath(op trace.Operation, deviceId string, target string) (*FileStat, error)
+}
+
+func StatPath(ctx context.Context, vc *exec.Container, path string) (*types.GuestFileInfo, error) {
+	defer trace.End(trace.Begin(vc.Config.Name))
+
+	// below implementation of searching one directory up does not work at root, but we can assume root is a directory
+	if path == "/" {
+		return &types.GuestFileInfo{
+			Path: "/",
+			Type: string(types.GuestFileTypeDirectory),
+			Size: int64(4096),
+		}, nil
+	}
+
+	filemgr, err := guest.NewOperationsManager(vc.VIM25Reference(), vc.VMReference()).FileManager(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	auth := types.NamePasswordAuthentication{
+		Username: vc.ExecConfig.ID,
+	}
+
+	// list files one directory up form the target path to include directories.
+	pathBasename := filepath.Base(path)
+	for _, path := range []string{path, strings.Replace(path, pathBasename, "", -1)} {
```

Comment:
rather than calling base and then replacing it with `""` you should just use `filepath.Split(path)` This will return the path to the parent of the target and then only the target string. e.g. filepath.Split("/my/example/target") -> "/my/example" and "target"